scanpy/readwrite.py

```diff
@@ -418,6 +422,26 @@ def read_visium(
             inplace=True,
         )
 
+        # try to find path to high-res tissue image
+        if tissue_image_path is None:
+            cand_image_paths = [
+                path / ("image" + ext) for ext in ['.tif', '.tiff', '.jpg', '.jpeg']
+            ]
+            cand_image_paths += [
+                path / (library_id + "_image" + ext)
+                for ext in ['.tif', '.tiff', '.jpg', '.jpeg']
+            ]
+            for p in cand_image_paths:
+                if p.exists():
+                    tissue_image_path = p.resolve()
+                    break
+        # put image path in uns
+        else:
+            tissue_image_path = Path(tissue_image_path).resolve()
```

Comment:
Is it very important that you guess where the image is? I would rather this not happen, since it could easily read the wrong thing, or not find an image a user would want to pick up. I'd rather the user had to be explicit about this.

Reply:
From a user perspective its more convenient if the path is assumed. Not that this is also the case for the png images (assumed to be in `spatial/tissue_hires_image.png`). This is the case for the standardised datasets you can download from visium but many of course be different for datasets that you create yourself. 
I do understand your point though, guessing is rarely a very good idea.

Reply:
I think there are two issues with this:

* If it's wrong, it's not obvious until later in the analysis. If there are other images in the directory, for example: low-res thumbnails or plots from the user's pipeline could get picked up by accident.
* As it stands right now, if a user ever passed `"include_tiff"` (or whatever we end up calling it), it will always be included on subsequent calls. I think it's bad for there to be state like that in these functions.

Reply:
ok, sure. Lets leave it at having the user passing `source_image_path` to `read_visium` when they want to include this information in the adata.